linera-views/src/backends/rocks_db.rs

```diff
@@ -291,35 +264,45 @@ impl WritableKeyValueStore for RocksDbStoreInternal {
             if let WriteOperation::DeletePrefix { key_prefix } = op {
                 if get_upper_bound(key_prefix) == Bound::Unbounded {
                     for short_key in self.find_keys_by_prefix(key_prefix).await? {
-                        let mut key = key_prefix.clone();
-                        key.extend(short_key);
-                        keys.push(key);
+                        let mut full_key = self.root_key.clone();
+                        full_key.extend(key_prefix);
+                        full_key.extend(short_key);
+                        keys.push(full_key);
                     }
                 }
             }
         }
         for key in keys {
             batch.operations.push(WriteOperation::Delete { key });
         }
+        let root_key = self.root_key.to_vec();
         tokio::task::spawn_blocking(move || -> Result<(), RocksDbStoreError> {
             let mut inner_batch = rocksdb::WriteBatchWithTransaction::default();
             for operation in batch.operations {
                 match operation {
                     WriteOperation::Delete { key } => {
                         ensure!(key.len() <= MAX_KEY_SIZE, RocksDbStoreError::KeyTooLong);
-                        inner_batch.delete(&key)
+                        let mut full_key = root_key.clone();
```

Comment:
use `root_key` directly?

Reply:
How could we do that?
If we concatenate the key in write, then we need to concatenate them in delete.

Reply:
I meant to spare a `clone()`: you already cloned it and moved it above.

Reply:
I guess just try `let mut full_key = root_key;`

Reply:
I had to clone it because otherwise, I could not access the variable in the `tokio::spawn`.
The operation `let mut full_key = root_key` is not possible without a copy of sort since there could be several delete operations. Since the `Vec<_>` does not implement the `Copy` trait, an explicit clone operation is needed. The compiler agrees with me.

What is unfortunate is that the construction of the inner batch only takes a reference, but as far as I know, it is not possible to extend a reference.